tools/cli/util.go

```diff
@@ -738,51 +738,71 @@ func newContextWithTimeout(c *cli.Context, timeout time.Duration) (context.Conte
 
 // process and validate input provided through cmd or file
 func processJSONInput(c *cli.Context) *commonpb.Payloads {
-	rawJson := processJSONInputHelper(c, jsonTypeInput)
-	if len(rawJson) == 0 {
-		return nil
-	}
+	jsonsRaw := readJSONInputs(c, jsonTypeInput)
+
+	var jsons []interface{}
+	for _, jsonRaw := range jsonsRaw {
+		if len(jsonRaw) == 0 {
+			continue
+		}
 
-	var v interface{}
-	if err := json.Unmarshal(rawJson, &v); err != nil {
-		ErrorAndExit("Input is not a valid JSON.", err)
+		var j interface{}
+		if err := json.Unmarshal(jsonRaw, &j); err != nil {
+			ErrorAndExit("Input is not a valid JSON.", err)
+		}
+		jsons = append(jsons, j)
 	}
 
-	p, err := payloads.Encode(v)
+	p, err := payloads.Encode(jsons...)
 	if err != nil {
 		ErrorAndExit("Unable to encode Input.", err)
 	}
 
 	return p
 }
 
-// process and validate json
-func processJSONInputHelper(c *cli.Context, jType jsonType) []byte {
-	var flagNameOfRawInput string
-	var flagNameOfInputFileName string
+// read multiple inputs presented in json format
+func readJSONInputs(c *cli.Context, jType jsonType) [][]byte {
+	var flagRawInput string
+	var flagInputFileName string
 
 	switch jType {
 	case jsonTypeInput:
-		flagNameOfRawInput = FlagInput
-		flagNameOfInputFileName = FlagInputFile
+		flagRawInput = FlagInput
+		flagInputFileName = FlagInputFile
 	case jsonTypeMemo:
-		flagNameOfRawInput = FlagMemo
-		flagNameOfInputFileName = FlagMemoFile
+		flagRawInput = FlagMemo
+		flagInputFileName = FlagMemoFile
 	default:
 		return nil
 	}
 
-	if c.IsSet(flagNameOfRawInput) {
-		return []byte(c.String(flagNameOfRawInput))
-	} else if c.IsSet(flagNameOfInputFileName) {
-		inputFile := c.String(flagNameOfInputFileName)
+	if c.IsSet(flagRawInput) {
+		inputsG := c.Generic(flagRawInput)
+
+		var inputs *cli.StringSlice
+		var ok bool
+		if inputs, ok = inputsG.(*cli.StringSlice); !ok {
+			// input could be provided as StringFlag instead of StringSliceFlag
+			ss := make(cli.StringSlice, 1)
+			ss[0] = fmt.Sprintf("%v", inputsG)
```

Comment:
this is weird. doesn't this mean its a cli.String type then? Why fmt.Sprintf it?

Reply:
`cli.String` doesn't exists in the api of urfave/cli. There is only a private field `cli.stringValue`

Reply:
for reference https://godoc.org/gopkg.in/urfave/cli.v1

Reply:
makes sense